core/include/c_api/tiledb.h

```diff
@@ -319,11 +317,115 @@ TILEDB_EXPORT int tiledb_basic_array_create(
     tiledb_ctx_t* ctx, const char* name);
 
 /* ********************************* */
-/*               ARRAY               */
+/*            ATTRIBUTE              */
 /* ********************************* */
 
-/** A TileDB array object. */
-typedef struct tiledb_array_t tiledb_array_t;
+/** A TileDB attribute. */
+typedef struct tiledb_attribute_t tiledb_attribute_t;
+
+/**
+ * Creates a TileDB attribute.
+ *
+ * @param name The attribute name.
+ * @param type The attribute type.
+ * @return The created TileDB attribute.
+ */
+TILEDB_EXPORT tiledb_attribute_t* tiledb_attribute_create(
+    const char* name, tiledb_datatype_t type);
+
+/**
+ * Destroys a TileDB attribute, freeing-up memory.
+ *
+ * @param attr The attribute to be destroyed.
+ * @return TILEDB_OK for success and TILEDB_ERR for error.
+ */
+TILEDB_EXPORT int tiledb_attribute_free(tiledb_attribute_t* attr);
+
+/**
+ * Sets a compressor to an attribute.
+ *
+ * @param attr The target attribute.
+ * @param compressor The compressor to be set.
+ * @return TILEDB_OK for success and TILEDB_ERR for error.
+ */
+TILEDB_EXPORT int tiledb_attribute_set_compressor(
+    tiledb_attribute_t* attr, tiledb_compressor_t compressor);
+
+/**
+ * Sets the compression level to an attribute.
+ *
+ * @param attr The target attribute.
+ * @param compression_level The compression level.
+ * @return TILEDB_OK for success and TILEDB_ERR for error.
+ */
+TILEDB_EXPORT int tiledb_attribute_set_compression_level(
```

Comment:
which compressor doesn't take a compression level parameter?  I think it would make sense to add this as an attribute of the `set_compressor` rather than it being another function.  we could have `-1` as a default compression level (don't care).